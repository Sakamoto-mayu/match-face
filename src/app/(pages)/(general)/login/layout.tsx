export const metadata = {
  title: "ログイン",
  description: "一般ユーザーログインページ",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
