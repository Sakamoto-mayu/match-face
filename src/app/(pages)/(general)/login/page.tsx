import Input from '@/components/ui/Input';
import OrangeButton from '@/components/ui/button/OrangeButton';
import Link from 'next/link';

const LoginPage = () => {
  return (
    <div className="flex flex-col">
      <h1 className=" text-orange font-normal">Match Face</h1>
      <label htmlFor="userId" className="">
        ユーザーID
      </label>
      <Input id="userId" />
      <label htmlFor="password" className="">
        パスワード
      </label>
      <Input id="password" />
      <OrangeButton label="ログイン" />
      <Link href="/remind" className="text-blue-500">
        パスワードを忘れた
      </Link>
      <Link href="/admin/login" className="text-blue-500">
        管理者ログイン
      </Link>
    </div>
  );
};

export default LoginPage;
