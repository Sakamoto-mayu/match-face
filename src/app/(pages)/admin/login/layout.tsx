export const metadata = {
  title: "ログイン",
  description: "管理者ログインページ",
};

export default function LoginLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
