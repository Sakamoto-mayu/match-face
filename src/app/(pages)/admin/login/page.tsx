'use client';
import Input from '@/components/ui/Input';
import SiteTitle from '@/components/ui/SiteTitle';
import OrangeButton from '@/components/ui/button/OrangeButton';
import users from '@/const/login';
import Link from 'next/link';
import { useRouter } from 'next/navigation';
import { FormEvent, useState } from 'react';

const AdminLoginPage = () => {
  const router = useRouter();
  const [userId, setUserId] = useState<string>();
  const [password, setPassword] = useState<string>();
  // const [userIdError, setUserIdError] = useState<boolean>(false);
  const [userIdBlankError, setUserIdBlankError] = useState<boolean>(false);
  // const [passwordError, setPasswordError] = useState<boolean>(false);
  const [passwordBlankError, setPasswordBlankError] = useState<boolean>(false);

  // ダミーデータ
  const adminCorrectUser = users[2];
  // console.log('adminCorrectUser', adminCorrectUser);

  // todo:（仮）ログイン認証チェック(入力欄がブランクの時のみエラー感知)
  const checkLogin = (event: FormEvent) => {
    event.preventDefault();
    if (!userId && !password) {
      setUserIdBlankError(true);
      setPasswordBlankError(true);
    } else if (!userId && password) {
      setUserIdBlankError(true);
      setPasswordBlankError(false);
    } else if (userId && !password) {
      setUserIdBlankError(false);
      setPasswordBlankError(true);
    } else if (
      userId === adminCorrectUser.user_id &&
      password === adminCorrectUser.password
    ) {
      router.push('/admin');
    }
  };

  return (
    <>
      <form onSubmit={checkLogin}>
        <div className="flex flex-col items-center justify-center h-screen">
          <div className="flex flex-col items-center m-3">
            <SiteTitle className="m-3" />
            <h2>管理者用ログイン</h2>
          </div>

          <div className="mt-2 mb-2">
            <label htmlFor="userId" className="">
              ユーザーID
            </label>
            <Input
              id="userId"
              className="w-96 h-8 mt-2"
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                setUserId(e.target.value)
              }
            />
            {userIdBlankError && (
              <p className="text-red">※ユーザーIDを入力してください</p>
            )}
          </div>
          <div>
            <label htmlFor="password" className="">
              パスワード
            </label>
            <Input
              id="password"
              className="w-96 h-8 mt-2"
              onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                setPassword(e.target.value)
              }
            />
            {passwordBlankError && (
              <p className="text-red">※パスワードを入力してください</p>
            )}
          </div>

          <OrangeButton
            label="ログイン"
            className="mt-10 mb-4 w-48 rounded-none"
            type="submit"
          />
          <Link href="/remind" className=" text-blue">
            パスワードを忘れた
          </Link>
          <Link href="/login" className="text-blue pt-16">
            一般ログイン
          </Link>
        </div>
      </form>
    </>
  );
};

export default AdminLoginPage;
