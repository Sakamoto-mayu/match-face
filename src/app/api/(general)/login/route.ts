import { cookies } from 'next/headers';
import { NextResponse } from 'next/server';

export async function POST(request: Request) {
  const res = await request.json();
  const userId = res.userId;
  const password = res.password;
  const response = await fetch(
    `http://localhost:8080/qa_system_api/users/${userId}?password=${password}`
  );
  const data = await response.json();
  cookies().set({
    name: 'userId',
    value: `${userId}`,
    httpOnly: true,
    secure: true,
    path: '/',
  });
  return NextResponse.json(data);
}
