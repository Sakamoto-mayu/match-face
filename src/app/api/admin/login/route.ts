import { cookies } from 'next/headers';
import { NextResponse } from 'next/server';

export async function POST(request: Request) {
  const res = await request.json();
  const administratorId = res.administratorId;
  const password = res.password;
  try {
    const response = await fetch(
      `${process.env.BE_URL}/administrators/${administratorId}?password=${password}`
    );
    const data = await response.json();
    cookies().delete('userId');
    if (data.administrator.id && data.administrator.length !== 0) {
      cookies().set({
        name: 'administratorId',
        value: `${data.administrator.id}`,
        httpOnly: true,
        secure: true,
        path: '/',
        maxAge: 60 * 60 * 24,
      });
    }
    return NextResponse.json(data);
  } catch (error) {
    throw new Error('api error');
  }
}
