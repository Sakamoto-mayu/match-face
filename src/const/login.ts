type User = {
  id: number;
  name: string;
  password: string;
  email: string;
  hire_date: string;
  department_id: number;
  status_id: number;
  created_user: string;
  created_at: string;
  update_user: string;
  update_at: string;
};

type Users = User[];

const users: Users = [
  {
    id: 1111,
    name: 'ラクス太郎',
    password: 'password#1',
    email: 'rakus@example.com',
    hire_date: '2023-01-01',
    department_id: 1,
    status_id: 1,
    created_user: 'ラクス責任者',
    created_at: '2022-12-30',
    update_user: 'ラクス責任者',
    update_at: '2023-01-02',
  },
  {
    id: 2222,
    name: 'ラクス次郎',
    password: 'password#2',
    email: 'rakus2@example.com',
    hire_date: '2023-01-01',
    department_id: 2,
    status_id: 2,
    created_user: 'ラクス責任者',
    created_at: '2022-12-30',
    update_user: 'ラクス責任者',
    update_at: '2023-01-02',
  },
  {
    id: 3333,
    name: 'ラクス三郎',
    password: 'password#3',
    email: 'rakus3@example.com',
    hire_date: '2023-01-01',
    department_id: 3,
    status_id: 3,
    created_user: 'ラクス責任者',
    created_at: '2022-12-30',
    update_user: 'ラクス責任者',
    update_at: '2023-01-02',
  },
];

export default users;
