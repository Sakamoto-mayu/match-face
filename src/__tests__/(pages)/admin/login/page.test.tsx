import AdminLoginPage from '@/app/(pages)/admin/login/page';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { AppRouterContextProviderMock } from '@/__tests__/test_utils/app-router-context-provider-mock';
import { admin, dummyAdmin } from '@/const/login';

const push = jest.fn();
const user = userEvent.setup();

// ダミーデータ
const registeredAdmin = admin[0];
const notRegisteredAdmin = dummyAdmin[0];

// describe('スナップショット', () => {
//   test('スナップショット', () => {
//     const { container } = render(
//       <AppRouterContextProviderMock router={{ push }}>
//         <AdminLoginPage />
//       </AppRouterContextProviderMock>
//     );
//     expect(container).toMatchSnapshot();
//   });
// });

describe('ページがレンダリングされた時', () => {
  beforeEach(() => {
    render(
      <AppRouterContextProviderMock router={{ push }}>
        <AdminLoginPage />
      </AppRouterContextProviderMock>
    );
  });

  // test('"パスワードを忘れた"のリンクのurlが"/remind"であること', () => {
  //   //"パスワードを忘れた"のリンク
  //   const remindLink = screen.getByRole('link', { name: 'パスワードを忘れた' });
  //   expect(remindLink).toHaveAttribute('href', '/remind');
  // });

  test('"一般ログイン"のリンクのurlが"/login"であること', () => {
    // "一般ログイン"のリンク
    const userLoginLink = screen.getByRole('link', { name: '一般ログイン' });
    expect(userLoginLink).toHaveAttribute('href', '/login');
  });
});

// describe('初めてOrangeButtonがクリックされた時', () => {
//   beforeEach(() => {
//     render(
//       <AppRouterContextProviderMock router={{ push }}>
//         <AdminLoginPage />
//       </AppRouterContextProviderMock>
//     );
//   });

//   describe('ユーザーIDが未入力の時', () => {
//     test('パスワードが未入力の場合、"※ユーザーIDを入力してください。"と、"※パスワードを入力してください。"が表示されること', async () => {
//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※ユーザーIDを入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//     test('登録済みのパスワードが入力されている場合、”※ユーザーIDを入力してください。”が表示されること', async () => {
//       // 登録済みのパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※ユーザーIDを入力してください。')
//       ).toBeInTheDocument();
//     });

//     test('登録されていないパスワードが入力されている場合、"※ユーザーIDを入力してください。"と、"正しいパスワードを入力してください。"が表示されること', async () => {
//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※ユーザーIDを入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//   });

//   describe('ユーザーIDが全角数字のみで入力されている時', () => {
//     test('パスワードが未入力の場合、"※半角数字で入力してください。"と、"※パスワードを入力してください"が表示されること', async () => {
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       // ユーザーIDを全角数字のみで入力
//       await userEvent.type(userIdInput, '１１１１');

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();
//     });

//     test('登録済みのパスワードが入力されている場合、"※半角数字で入力してください。"が表示されること', async () => {
//       // ユーザーIDを全角数字のみで入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, '１１１１');

//       // 登録済みのパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//     });
//     test('登録されていないパスワードが入力されている場合、"※半角数字で入力してください。"と、”正しいパスワードを入力してください。"が表示されること', async () => {
//       // ユーザーIDを全角数字のみで入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, '１１１１');

//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//   });

//   describe('ユーザーIDが半角英語のみで入力されている時', () => {
//     test('パスワードが未入力の場合、"※半角数字で入力してください。"と、"※パスワードを入力してください"が表示されること', async () => {
//       // ユーザーIDを半角英語のみで入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, 'aaaa');

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();
//     });

//     test('登録済みのパスワードが入力されている場合、"※半角数字で入力してください。"が表示されること', async () => {
//       // ユーザーIDを半角英語のみで入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, 'aaaa');

//       // 登録済みのパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//     });

//     test('登録されていないパスワードが入力されている場合、"※半角数字で入力してください。"と、”正しいパスワードを入力してください。"が表示されること', async () => {
//       // ユーザーIDを半角英語のみで入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, 'aaaa');

//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//   });

//   describe('ユーザーIDが半角英語と半角数字で入力されている時', () => {
//     test('パスワードが未入力の場合、"※半角数字で入力してください。"と、"※パスワードを入力してください"が表示されること', async () => {
//       // ユーザーIDを半角英語と半角数字で入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, '111a');

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();
//     });

//     test('登録済みのパスワードが入力されている場合、"※半角数字で入力してください。"が表示されること', async () => {
//       // ユーザーIDを半角英語と半角数字で入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, '111a');

//       // 登録済みのパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//     });

//     test('登録されていないパスワードが入力されている場合、"※半角数字で入力してください。"と、”正しいパスワードを入力してください。"が表示されること', async () => {
//       // ユーザーIDを半角英語と半角数字で入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, '111a');

//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//   });

//   describe('登録されているユーザーIDが入力されている時', () => {
//     test('パスワードが未入力の場合、"※パスワードを入力してください"が表示されること', async () => {
//       // 登録されているユーザーIDを入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//     test('登録済みのパスワードが入力されている場合、ログインに成功し管理者トップ画面に遷移すること', async () => {
//       // 登録済みユーザーIDを入力;
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // 登録済みパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // 管理者トップ画面へ遷移
//       expect(push).toHaveBeenCalledWith('/admin');
//     });
//     test('登録されていないパスワードが入力されている場合、”正しいパスワードを入力してください。"が表示されること', async () => {
//       // 登録済みユーザーIDを入力;
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//   });

//   describe('ユーザーIDが半角数字のみで入力されているが、登録されていない時', () => {
//     test('パスワードが未入力の場合、"※正しいユーザーIDを入力してください。"と、"※パスワードを入力してください"が表示されること', async () => {
//       // 半角数字のみだが登録されていないユーザーIDを入力(99999)
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${notRegisteredAdmin.id}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※正しいユーザーIDを入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//     test('登録済みのパスワードが入力されている場合、"※正しいユーザーIDを入力してください。"が表示されること', async () => {
//       // 半角数字のみだが登録されていないユーザーIDを入力(99999)
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${notRegisteredAdmin.id}`);

//       // 登録済みパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※正しいユーザーIDを入力してください。')
//       ).toBeInTheDocument();
//     });
//     test('登録されていないパスワードが入力されている場合、"※正しいユーザーIDを入力してください。"と、正しいパスワードを入力してください。”が表示されること', async () => {
//       // 半角数字のみだが登録されていないユーザーIDを入力(99999)
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${notRegisteredAdmin.id}`);

//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // ボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示される
//       expect(
//         await screen.findByText('※正しいユーザーIDを入力してください。')
//       ).toBeInTheDocument();
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();
//     });
//   });
// });

// describe('OrangeButtonがクリックされた時（2回目以降）', () => {
//   beforeEach(() => {
//     render(
//       <AppRouterContextProviderMock router={{ push }}>
//         <AdminLoginPage />
//       </AppRouterContextProviderMock>
//     );
//   });

//   describe('"※ユーザーIDを入力してください。"のみが表示されている時', () => {
//     test('登録済みのパスワードと登録済みのユーザーIDが入力されている場合、"※ユーザーIDを入力してください。"の表示が消え、ログインに成功しトップ画面に遷移すること', async () => {
//       // 登録済みパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // 初めてボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示されている
//       expect(
//         await screen.findByText('※ユーザーIDを入力してください。')
//       ).toBeInTheDocument();

//       // 登録済みユーザーIDを入力;
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // ボタンをクリック（2回目）
//       await user.click(loginButton);

//       // エラーメッセージが消えたことを確認
//       expect(
//         await waitFor(() =>
//           screen.queryByText('※ユーザーIDを入力してください。')
//         )
//       ).toBeNull();

//       // トップ画面へ遷移
//       expect(push).toHaveBeenCalledWith('/admin');
//     });
//   });

//   describe('"※半角数字で入力してください"のみが表示されている時', () => {
//     test('登録済みのパスワードと登録済みのユーザーIDが入力されている場合、"※半角数字で入力してください。"の表示が消え、ログインに成功しトップ画面に遷移すること', async () => {
//       // ユーザーIDを半角英語で入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, 'aaaa');

//       // 登録済みパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // 初めてボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示されている
//       expect(
//         await screen.findByText('※半角数字で入力してください。')
//       ).toBeInTheDocument();

//       // 入力した値を削除
//       await userEvent.clear(userIdInput);
//       // 登録済みユーザーIDを入力;
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // ボタンをクリック（2回目）
//       await user.click(loginButton);

//       // エラーメッセージが消えたことを確認
//       expect(
//         await waitFor(() => screen.queryByText('※半角数字で入力してください。'))
//       ).toBeNull();

//       // トップ画面へ遷移
//       expect(push).toHaveBeenCalledWith('/admin');
//     });
//   });

//   describe('"※正しいユーザーIDを入力してください。"のみが表示されている時', () => {
//     test('登録済みのパスワードと登録済みのユーザーIDが入力されている場合、"※正しいユーザーIDを入力してください。"の表示が消え、ログインに成功しトップ画面に遷移すること', async () => {
//       // 登録されていないユーザーIDを入力
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${notRegisteredAdmin.id}`);

//       // 登録済みパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // 初めてボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示されている
//       expect(
//         await screen.findByText('※正しいユーザーIDを入力してください。')
//       ).toBeInTheDocument();

//       // 入力した値を削除
//       await userEvent.clear(userIdInput);
//       // 登録済みユーザーIDを入力;
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // ボタンをクリック（2回目）
//       await user.click(loginButton);

//       // エラーメッセージが消えたことを確認
//       expect(
//         await waitFor(() =>
//           screen.queryByText('※正しいユーザーIDを入力してください。')
//         )
//       ).toBeNull();

//       // トップ画面へ遷移
//       expect(push).toHaveBeenCalledWith('/admin');
//     });
//   });

//   describe('"※パスワードを入力してください"のみが表示されている時', () => {
//     test('登録済みのユーザーIDと登録済みのパスワードが入力されている場合、"※パスワードを入力してください。"の表示が消え、ログインに成功しトップ画面に遷移すること', async () => {
//       // 登録済みユーザーIDを入力;
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // 初めてボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示されている
//       expect(
//         await screen.findByText('※パスワードを入力してください。')
//       ).toBeInTheDocument();

//       // 登録済みパスワードを入力;
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック（2回目）
//       await user.click(loginButton);

//       // エラーメッセージが消えたことを確認
//       expect(
//         await waitFor(() =>
//           screen.queryByText('※パスワードを入力してください。')
//         )
//       ).toBeNull();

//       // トップ画面へ遷移
//       expect(push).toHaveBeenCalledWith('/admin');
//     });
//   });

//   describe('"※正しいパスワードを入力してください。"のみが表示されている時', () => {
//     test('登録済みのユーザーIDと登録済みのパスワードが入力されている場合、"※正しいパスワードを入力してください。"の表示が消え、ログインに成功しトップ画面に遷移すること', async () => {
//       // 登録済みユーザーIDを入力;
//       const userIdInput = screen.getByLabelText('ユーザーID');
//       await userEvent.type(userIdInput, `${registeredAdmin.id}`);

//       // 登録されていないパスワードを入力
//       const passwordInput = screen.getByLabelText('パスワード');
//       await userEvent.type(passwordInput, `${notRegisteredAdmin.password}`);

//       // 初めてボタンをクリック
//       const loginButton = screen.getByRole('button', { name: 'ログイン' });
//       await user.click(loginButton);

//       // エラーメッセージが表示されている
//       expect(
//         await screen.findByText('※正しいパスワードを入力してください。')
//       ).toBeInTheDocument();

//       // 入力した値を削除
//       await userEvent.clear(passwordInput);
//       // 登録済みパスワードを入力;
//       await userEvent.type(passwordInput, `${registeredAdmin.password}`);

//       // ボタンをクリック（2回目）
//       await user.click(loginButton);

//       // エラーメッセージが消えたことを確認
//       expect(
//         await waitFor(() =>
//           screen.queryByText('※正しいパスワードを入力してください。')
//         )
//       ).toBeNull();

//       // トップ画面へ遷移
//       expect(push).toHaveBeenCalledWith('/admin');
//     });
//   });
// });
