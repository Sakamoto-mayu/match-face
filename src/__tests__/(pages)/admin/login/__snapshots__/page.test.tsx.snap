// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット スナップショット 1`] = `
<div>
  <form>
    <div
      class="flex flex-col items-center justify-center h-screen"
    >
      <div
        class="flex flex-col items-center m-3"
      >
        <h1
          class="text-orange text-4xl font-semibold m-3"
        >
          Match Face
        </h1>
        <h2>
          管理者用ログイン
        </h2>
      </div>
      <div
        class="mt-2 mb-2"
      >
        <label
          class=""
          for="administratorId"
        >
          ユーザーID
          <div>
            <input
              class="w-96 h-10 mt-2 border border-black"
              id="administratorId"
              name="administratorId"
              type="text"
            />
          </div>
        </label>
      </div>
      <div>
        <label
          class=""
          for="password"
        >
          パスワード
          <div>
            <input
              class="w-96 h-10 mt-2 border border-black"
              id="password"
              name="password"
              type="password"
            />
          </div>
        </label>
      </div>
      <div>
        <button
          class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white p-2 h-auto text-lg mt-10 mb-4 w-48 rounded-none"
          type="submit"
        >
          ログイン
        </button>
      </div>
      <p
        class="text-blue"
      >
        パスワードを忘れた
      </p>
      <a
        class="text-blue pt-16"
        href="/login"
      >
        一般ログイン
      </a>
    </div>
  </form>
</div>
`;
